### DISCUSSION TITLE
Cloudwatch: Add support for resolution of resource tags in alias

### DISCUSSION THREAD

**Discussion by Alex-Richman** [grafana/grafana#41816] [Data Sources]
Add support for using resource tags in the CloudWatch query alias.

See attached PR for details.

Example:
![2](https://user-images.githubusercontent.com/1280519/91644580-416c3c00-ea35-11ea-867a-61fbcb02be7a.png)

**Comment by kmoberg** [comment]
This _should_ be a no-brainer. Presenting instance IDs that change constantly, makes no sense, when we all tag the resources that help us show what they actually are.

**Comment by macobo** [comment]
In https://github.com/grafana/grafana/pull/27281 @marefr said that:

> However, the referenced feature request #27280 haven't seen any activity/comments/upvotes, that together with the extra complexity, maintenance burden and possibly cost (performance and calls) makes it hard for us to accept this currently. 

Here's a list of related or the same issue requests:
- https://github.com/grafana/grafana/pull/8307
- https://github.com/monitoringartist/grafana-aws-cloudwatch-dashboards/issues/14
- https://github.com/grafana/grafana/issues/13643
- https://github.com/grafana/grafana/issues/7294
- https://github.com/monitoringartist/grafana-aws-cloudwatch-dashboards/issues/41
- https://community.grafana.com/t/cloudwatch-tags-as-alias/39931
- https://community.grafana.com/t/retrieve-values-for-tag-in-cloudwatch/9098
- https://community.grafana.com/t/seeking-to-retrieve-aws-cloudwatch-disk-volume-tags/7878
- https://rtfm.co.ua/en/prometheus-yet-another-cloudwatch-exporter-collecting-aws-cloudwatch-metrics/

See also comments under the original PR after it got closed.

People feel pretty strongly about this issue because cloudwatch is pretty weak on grafana without it. After all, what use are metrics without being able to distinguish or filter _what_ data is being displayed.

**Comment by DerfOh** [comment]
2 years later and no solutions or workarounds? Are people out there just memorizing hundreds of IDs or has there been a solution posted that I cannot find? Getting tag values seems to work for variables but hits the fan if you're doing templating or multi-variable dashboard panels. 

Using the variable is workable if you only want to see one series at a time but when trying to do any comparisons between metrics it's manual method with verbose IDs or bust. Seeing how Amazon even has their own managed grafana product it seems that including tags is kind of a massive oversite, what good is cloudwatch integration if it's a massive headache to filter for relevant data? 

I'm having a hard time not passing judgment on grafana right now as it seems like there's a huge amount of discussion and requests for this feature but these discussions have been fragmented as a result of some sort of effort to close PRs by the grafana team, to make matters worse they brush it off stating there isn't enough of a need or a discussion around this topic to justify the development and maintenance efforts. To that I say just google it, there is no shortage of results that have gone unanswered or have been directed to the issues @Alex-Richman has already linked to.

**Comment by sarahzinger** [comment]
Hi Folks,

As part of our work to move from [discussions to issues](https://github.com/grafana/grafana/discussions/73424) I've opened this [issue](https://github.com/grafana/grafana/issues/77953). We know this is a very frequently requested feature and it's definitely an issue we hope to address someday, but unfortunately we do not have a current timeline around it. Thank you for the detailed description and linking of relevant/similar issues and prs and discussions on this, we really appreciate it! This kind of work makes it much easier to take on whenever the time comes. 

As a potential workaround until we can address this issue properly in the cloudwatch datasource, I believe the [Cloudwatch Integration](https://grafana.com/docs/grafana-cloud/monitor-infrastructure/aws/cloudwatch-metrics/#cloudwatch-metrics) which stores metrics into prometheus also has the option to ingest tags, which might solve this for some Grafana Cloud customers, but still, we'd like to add tags support as well for the cloudwatch datasource. 

Thank you for your patience, please feel free to share any other suggestions on this topic in [this github issue](https://github.com/grafana/grafana/issues/77953).
